### PR TITLE
fix: NaN for Bar y and height attributes if data is null

### PR DESCRIPTION
**Pull request by ckifer**
## Description

If bar data is `null` then bar `y` and `height` value come back as `NaN` due to some bad optional handling.

This PR attempts to add good default values if this is the case.

(will consider this a draft but feel free to review)

## Related Issue

#3344

## Screenshots (if appropriate):

image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: demo/component/BarChart.tsx [hunk 1]
```diff
@@ -126,40 +127,36 @@ const pageData = [
   { name: 'Page G', uv: 189, pv: 4800, amt: 2400 },
 ];
 
-const RenderLabel = (props: any) => {
```
Comment (nikolasrieble) [inline demo/component/BarChart.tsx:129]: I would suggest to keep the diff in a fix PR as small as possible. This seems to be an unrelated cleanup.

File: src/cartesian/Bar.tsx [hunk 2]
```diff
@@ -226,7 +227,7 @@ export class Bar extends PureComponent<Props, State> {
         x,
         y,
         width,
-        height,
+        height: Number.isNaN(height) ? 0 : height,
```
Comment (nikolasrieble) [inline src/cartesian/Bar.tsx:230]: This line really is the only relevant change, right? The rest is just refactoring.

Comment (nikolasrieble) [reply src/cartesian/Bar.tsx:230]: The height is also used for the Tooltip position a few lines below this one. We should move `const validHeight = Number.isNaN(height) ? 0 : height` up a few lines and then use in both places in the return type.

**Review by nikolasrieble — CHANGES_REQUESTED** [review]
Lets use the validHeight also for the Tooltip

File: demo/component/BarChart.tsx (hunk 1 above)
Comment (ckifer) [reply demo/component/BarChart.tsx:129]: yeah sorry I couldn't help myself - I can't stand all the errors in these demo files LOL

File: src/cartesian/Bar.tsx (hunk 2 above)
Comment (ckifer) [reply src/cartesian/Bar.tsx:230]: line 194 is actually more relevant. 

line 194:
If we don't have a currentValueScale (aka, current value is null or undefined) then use the baseValueScale (aka whatever y value is computed at first without taking the value into account, and if we don't have that then use undefined and allow `NaN` down the line because something is wrong.

Comment (ckifer) [reply src/cartesian/Bar.tsx:230]: will check the tooltip comment out, I'm not sure

Comment (ckifer) [reply src/cartesian/Bar.tsx]: It turns out that height does not affect the tooltip because when the tooltip isn't shown (since there's no data) these values aren't used. But better safe than sorry so moved the check up.

**Comment by ckifer** [comment]
Requested changes made, this should be good to go

**Review by nikolasrieble — APPROVED** [review]
Thank you!
